Observer/TescoPaymentObserver.php

```diff
@@ -0,0 +1,89 @@
+<?php
+namespace Omise\Payment\Observer;
+
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Framework\Event\Observer;
+
+class TescoPaymentObserver implements ObserverInterface
+{
+    /**
+    * @var \Magento\Framework\App\Config\ScopeConfigInterface
+    */
+    private $_scopeConfig;    
+    
+    /**
+    * @var  \Magento\Framework\Mail\Template\TransportBuilder
+    */
+    private $_transportBuilder;
+    
+    /**
+    * @var  \Omise\Payment\Helper
+    */
+    private $_helper;
+
+    public function __construct(
+        \Omise\Payment\Helper\OmiseHelper $helper,
+        \Magento\Framework\Mail\Template\TransportBuilder $transportBuilder,
+        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
+    ) {
+        $this->_scopeConfig      = $scopeConfig;
+        $this->_helper           = $helper;
+        $this->_transportBuilder = $transportBuilder;
+    }
+
+    /**
+     *
+     * @param \Magento\Framework\Event\Observer $observer
+     * @return $this
+     */
+    public function execute(Observer $observer)
+    {
+        $order       = $observer->getEvent()->getOrder();
+        $payment     = $order->getPayment();
+        
+        if (!$payment) {
+            return $this;
+        }
+
+        $paymentData = $payment->getData();
+
+        if ($paymentData['additional_information']['payment_type'] !== 'bill_payment_tesco_lotus') {
+            return $this;
+        }
+
+        $amount        = number_format($paymentData['amount_ordered'], 2) . ' ' . $order->getOrderCurrency()->getCurrencyCode();
+        $barcodeHtml   = $this->_helper->convertTescoSVGCodeToHTML($paymentData['additional_information']['barcode']);
+        $storeName     = $this->_scopeConfig->getValue('trans_email/ident_sales/name', \Magento\Store\Model\ScopeInterface::SCOPE_STORE);
+        $storeEmail    = $this->_scopeConfig->getValue('trans_email/ident_sales/email', \Magento\Store\Model\ScopeInterface::SCOPE_STORE);
+        $customerEmail = $order->getCustomerEmail();
+        $orderId       = $order->getIncrementId();
+
+        // make sure timezone is Thailand.
+        date_default_timezone_set("Asia/Bangkok");
+
+        // get DateTime deadline that is in next 24 hours.
+        $validUntil    = date("d-m-Y H:i:s" , time() + 24 * 60 * 60) . ' ICT';
```

Comment:
@jacstn (code styling)  pls remove 1 space before `,`
```php
date("d-m-Y H:i:s", time() + 24 * 60 * 60) . ' ICT';
```

Reply:
@jacstn (note) I usually used to to the way of using 
```php
$datetime = new \DateTime('now', new \DateTimeZone('Asia/Bangkok'));
$datetime->add(new \DateInterval('P1D'));
// or
// $datetime->modify('+1 day');

$validUntil = $datetime->format('d-m-Y H:i:s') . ' ICT';
```

Reply:
@jacstn Also not sure but what do you think if we make it more in a human-friendly format?
like `format('M d, Y h:i A') . ' (GMT+7)'`
> Nov 20, 2018 12:33 PM (GMT+7)

Reply:
Month, day ... is US friendly.. entire world is using:
Day Month Year ?

Reply:
ok.. i understand what you mean... it will be in english? localization would be required to show month in thai language..

Reply:
i mean.. `Jul` instead of Thai language month... is that ok?

Reply:
@jacstn  Uhmm.. this might about UX thing, I'm not familiar with this. Let's see how Omise Dashboard implemented?

```
EN: September 20, 2018

TH: 20 พฤศจิกายน 2018
```

How about this ☝️ ?

Reply:
or can use the same format that Magento uses as well